src/Microsoft.Azure.WebJobs.Host.Storage/Singleton/BlobLeaseDistributedLockManager.cs

```diff
@@ -167,10 +167,21 @@ protected virtual BlobContainerClient GetParentBlobContainerClient(BlobClient bl
             bool blobDoesNotExist = false;
             try
             {
-                // Optimistically try to acquire the lease. The blob may not yet
-                // exist. If it doesn't we handle the 404, create it, and retry below
-                var leaseResponse = await GetBlobLeaseClient(blobClient, proposedLeaseId).AcquireAsync(leasePeriod, cancellationToken: cancellationToken);
-                return leaseResponse.Value.LeaseId;
+                // Check if a lease is available before trying to acquire. The blob may
+                // not yet exist; if it doesn't we handle the 404, create it, and retry below
+                var blobProperties = await ReadLeaseBlobMetadata(blobClient, cancellationToken);
+
+                switch (blobProperties?.LeaseState)
```

Comment:
Can you add a comment here explaining why we're doing it this way?

Reply:
Can do!